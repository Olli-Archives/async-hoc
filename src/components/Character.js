import React from 'react';
import PropTypes from 'prop-types';

export default function Character({ name, status, species, image }){
  return (
    <div>
      <h1>{name}</h1>
      <p>{status}</p>
      <p>{species}</p>
      <img src={image}/>
    </div>
  );

}
Character.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired 
};
