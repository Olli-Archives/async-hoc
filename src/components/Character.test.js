import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character Component', ()=>{
  it('renders a character with name, status, species, and image ', ()=>{
    const wrapper = shallow(<Character name='rick' status='alive' species='human' image='test-image'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
