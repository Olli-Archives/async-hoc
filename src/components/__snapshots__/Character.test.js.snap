// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component renders a character with name, status, species, and image  1`] = `
<div>
  <h1>
    rick
  </h1>
  <p>
    alive
  </p>
  <p>
    human
  </p>
  <img
    src="test-image"
  />
</div>
`;
